pkg/imageverifycache/interface.go

```diff
@@ -0,0 +1,17 @@
+package imageverifycache
+
+import "context"
+
+type Client interface {
+	// Set Adds an image to the cache. The image is considered to be verified for the given rule in the policy
+	// The entry outomatically expires after sometime
+	// Returns true when the cache entry is added
+	Set(ctx context.Context, policyId string, policyVersion string, ruleName string, imageRef string) (bool, error)
```

Comment:
I wonder if it would make sense to pass the policy interface rather than a policy id and resource version 🤔

Reply:
@eddycharly If we pass the rule name and image ref before and evaluate the policy version afterwards (when we call the `Set()`). Isn't it possible that the policy has been modified while we are setting the cache. So we will be adding an entry with the new policy version but the rule name and image ref are from previous version

Maybe we should also pass the Rule and do a check if the rule is still in the policy.

Reply:
nevermind I think version info are stored and not evaluated so they cannot change